server/controllers/finance/reports/vouchers/index.js

```diff
@@ -126,3 +135,18 @@ function report(req, res, next) {
     .done();
 }
 
+function creditNotedRef(uuid) {
+  const sql = `
+    SELECT BUID(v.uuid) as uuid, v.date, v.project_id, v.currency_id, v.amount,
+      v.description, v.user_id, v.type_id,  u.display_name, transaction_type.text,
+      CONCAT_WS('.', '${entityIdentifier}', p.abbr, v.reference) AS reference, reversed
+    FROM voucher v
+    JOIN project p ON p.id = v.project_id
```

Comment:
All these JOINs are expensive, but I'm not sure we need them.  In fact, I'm pretty sure we could just look this up in the document_map table.  Could we do this instead:
```js
const sql = `
  SELECT dm.text FROM document_map dm WHERE dm.uuid = ?;
`
return db.one(sql, db.bid(uuid));

Reply:
Otherwise, we could just do a `Vouchers.lookupVoucher(uuid)`.

Reply:
this sql query is not correct, here we are looking for the voucher whitch was created in order to canced the current one(the one we know it uuid)